src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1591 @@
+				if (solution_map_sizeX != solution_map_sizeY)
+				{
+					ShowWarningError("Error in solution space mapping, suggest adjusting operating constraints.");
+					return -2;//error
+				}
+				// Check that in this mode the //Outside Air Relative Humidity(0 - 100 % )	//Outside Air Humidity Ratio(g / g)//Outside Air Temperature(�C)
+				if (pMode->MeetsOAEnvConstraints(StepIns.Tosa, Wosa, 100 * StepIns.RHosa) == true)//fix this it should not end if it does not meet contraints.
+				{
+					EnvironmentConditionsMet = EnvironmentConditionsMetOnce = true;
+				}
+				else
+				{
+					EnvironmentConditionsMet = false;
+					if (!WarmupFlag)
+					{
+						//ShowWarningError("Failed to meet environment conditions for mode: " + RoundSigDigits(double(pMode->ModeID), 1));
+					}
+				}
+
+				if (EnvironmentConditionsMet == true)
+				{
+					double NormalizationReference = GetNormalPoint(1);// assumes the model has at least 1 curve and that all are set the same.
+
+					for (point_number = 0; point_number != solution_map_sizeX; point_number++) // within each mode go though all the combinations of solution spaces.
+					{
+						//Supply Air Mass Flow Rate(kg / s)
+						//Outside Air Fraction(0 - 1)
+						CSetting* pCandidateSetting = new CSetting;
+						MsaRatio = solutionspace->PointX[point_number];// fractions of rated mass flow rate, so for some modes this might be low but others hi
+						OSAF = solutionspace->PointY[point_number];
+						// there is no operating condition test, becuase it uses those to make the map so only allowed values are here.
+						Msa = NormalizationReference * MsaRatio;
+						//Calculate the ventilation mass flow rate
+						Mvent = Msa * OSAF;
+
+						if (StdRhoAir>1) 	pCandidateSetting->Supply_Air_Ventilation_Volume = Mvent / StdRhoAir;
+						else pCandidateSetting->Supply_Air_Ventilation_Volume = Mvent / 1.225;  //Why store it as a volume
+
+						
+						if (Mvent > MinOA_Msa) MinVRMet = MinVRMetOnce = true;
+						else MinVRMet = false;
+
+						if (MinVRMet)
+						{
+							//all these points meet the minimum VR requirement
+							solutionspace->PointMeta[point_number] = 1;
+							// Calculate prospective supply air temperature
+							Tsa = pMode->CalculateCurveVal(1, StepIns.Tosa, Wosa, StepIns.Tra, Wra, Msa, OSAF, modenumber, TEMP_CURVE); //TEMP_CURVE W_CURVE POWER_CURVE
+																														// Check it meets constraints
+							if (MeetsSupplyAirTOC(Tsa)) SAT_OC_Met = SAT_OC_MetOnce = SAT_OC_MetinMode = true;
+							else SAT_OC_Met = false;
+
+							// Calculate prospective supply air Humidity Ratio
+							Wsa = pMode->CalculateCurveVal(1, StepIns.Tosa, Wosa, StepIns.Tra, Wra, Msa, OSAF, modenumber, W_CURVE);
+							//Return Air Relative Humidity(0 - 100 % ) //Return Air Humidity Ratio(g / g)
+							if (MeetsSupplyAirRHOC(Wsa)) SARH_OC_Met = SAHR_OC_MetOnce = SAHR_OC_MetinMode = true;
+							else SARH_OC_Met = false;
+
+
+							if (SARH_OC_Met == true && SAT_OC_Met == true)
```

Comment:
No.  We don't need checks against `true`.  It is likely worthwhile to rename these variables so that the check for truthiness is obvious from the name.